src/types/api/main.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export type DesktopAPI = {
+
+    // Initialization
+    isDev: () => Promise<boolean>;
+    getAppInfo: () => Promise<{name: string; version: string}>;
+    reactAppInitialized: () => void;
+
+    // Session
+    setSessionExpired: (isExpired: boolean) => void;
+    onUserActivityUpdate: (listener: (
+        userIsActive: boolean,
+        idleTime: number,
+        isSystemEvent: boolean,
+    ) => void) => void;
+
+    // Unreads/mentions/notifications
+    sendNotification: (title: string, body: string, channelId: string, teamId: string, url: string, silent: boolean, soundName: string) => void;
+    onNotificationClicked: (listener: (channelId: string, teamId: string, url: string) => void) => void;
```

Comment:
Yeah let's not overengineer here, we'll just keep it as is.

The `url` came from the fact that I think something we get permalinks when popping notifications? I forget exactly. IMO, we should always just be using the `url` or the `channelId`/`teamId` pair, but I think that's a refinement to make at a later date.

Reply:
I suppose it should always be a URL ideally since the web app should know what that URL is when it sends the notification, but I suppose it could change in the future if we ever tie the RHS to the address like we've discussed doing in the past. But, yeah, we can definitely look at that later.